### PR TITLE
Update dependencies (and remove workaround reflection)

### PR DESCRIPTION
**Pull request by jskeet**
We should do a common protos 2.6.0 release before the GAX v4 release.

### CODE REVIEW

File: Google.Api.CommonProtos/Google.Api.CommonProtos.csproj [hunk 1]
```diff
@@ -14,7 +14,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Google.Protobuf" Version="[3.18.0, 4.0.0)" />
+    <PackageReference Include="Google.Protobuf" Version="[3.20.0, 4.0.0)" />
```
Comment (amanda-tarafa) [inline Google.Api.CommonProtos/Google.Api.CommonProtos.csproj]: Should this one be 3.21.0 ?? On https://github.com/googleapis/gapic-generator-csharp/pull/487 you said we should move to 3.21.0 before GaxV4

File: generateprotos.sh [hunk 2]
```diff
@@ -10,8 +10,8 @@ OS=windows
 declare -r ROOT=$(realpath $(dirname $0))
 cd $ROOT
 
-PROTOBUF_VERSION=3.18.0
-GRPC_VERSION=2.39.1
+PROTOBUF_VERSION=3.20.0
```
Comment (amanda-tarafa) [inline generateprotos.sh]: And here as well.

**Review by amanda-tarafa — APPROVED** [review]
LGTM. With a question

File: Google.Api.CommonProtos/Google.Api.CommonProtos.csproj (hunk 1 above)
Comment (jskeet) [reply Google.Api.CommonProtos/Google.Api.CommonProtos.csproj]: Hmm... had thought I'd done that. (It all got a bit confusing due to timing.) Will have a look.

**Pull request merged**